src/ConnectedMode/StatefulServerBranchProvider.cs

```diff
@@ -58,12 +66,26 @@ private void OnPreSolutionBindingUpdated(object sender, EventArgs e)
         {
             logger.LogVerbose(Resources.StatefulBranchProvider_BindingUpdated);
             selectedBranch = null;
+
+            threadHandling.RunOnBackgroundThread(NotifySlCoreBranchChange).Forget();
         }
 
         private void OnPreSolutionBindingChanged(object sender, ActiveSolutionBindingEventArgs e)
         {
             logger.LogVerbose(Resources.StatefulBranchProvider_BindingChanged);
             selectedBranch = null;
+
+            threadHandling.RunOnBackgroundThread(NotifySlCoreBranchChange).Forget();
+        }
+
+        private void NotifySlCoreBranchChange()
+        {
+            if (!serviceProvider.TryGetTransientService(out ISonarProjectBranchSlCoreService sonarProjectBranchSlCoreService))
+            {
+                logger.LogVerbose(SLCoreStrings.ServiceProviderNotInitialized);
+                return;
+            }
+            sonarProjectBranchSlCoreService.DidVcsRepositoryChange(new DidVcsRepositoryChangeParams(activeConfigScopeTracker.Current.Id));
```

Comment:
Race condition with ConfigScopeUpdater, not sure if the impact is big. Could be hard to fix